test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -80,12 +88,63 @@ spec = Test.group "Meta-Value Manipulation" <|
         Meta.is_an err Error . should_be_true
         Meta.is_a err Text . should_be_false
 
+        Meta.is_a Date.now Date.Date . should_be_true
+        Meta.is_a Date_Time.now Date_Time.Date_Time . should_be_true
+        Meta.is_a Date_Time.now Date.Date . should_be_false
+        Meta.is_a Time_Of_Day.now Time_Of_Day.Time_Of_Day . should_be_true
+        Meta.is_a Time_Of_Day.now Date.Date . should_be_false
+        Meta.is_a Date_Time.now.zone Time_Zone.Time_Zone . should_be_true
+        Meta.is_a Date_Time.now.zone Date.Date . should_be_false
+
+
+    Test.specify "should allow for returning the type of a value" <|
+        Meta.type_of 42 . should_equal Integer
+        Meta.type_of 2.81 . should_equal Decimal
+        Meta.type_of 2.81 . should_not_equal Integer
+
+        # Can't compare like
+        # Meta.type_of [1,2,3] . should_equal Vector.Vector
+        # because equality check tries to compare lengths, where RHS is a type
```

Comment:
I think we should just fix the definition of equality here.

I'm very unsure about the `is_of_type`. What is preventing us from fixing type equality?

Alternatively, can we rename `is_of_type` to something like `Meta.are_types_equal`?

Reply:
`is_of_type` sounds like `java.lang.Class.isInstance` check. While what it does is `java.lang.Class.isAssignable`!

When implementing _Array&Vector_ I also had problems with `==` and `should_equal`. I managed to solve them without introducing new API method. What are the core reasons why we cannot have `==` working properly on types - types are unique, right? E.g. Enso `==` is Java `==` on them, isn’t it?